packages/tspd/src/gen-extern-signatures/gen-extern-signatures.ts

```diff
@@ -2,32 +2,74 @@ import {
   CompilerHost,
   Decorator,
   Diagnostic,
+  type PackageJson,
   Program,
+  type SourceLocation,
   compile,
   createDiagnosticCollector,
+  createSourceFile,
   getLocationContext,
   getTypeName,
   joinPaths,
   navigateProgram,
   resolvePath,
-  type PackageJson,
 } from "@typespec/compiler";
 import prettier from "prettier";
+import { createDiagnostic } from "../ref-doc/lib.js";
 import { generateSignatureTests, generateSignatures } from "./decorators-signatures.js";
 import { DecoratorSignature } from "./types.js";
 
+function createSourceLocation(path: string): SourceLocation {
+  return { file: createSourceFile("", path), pos: 0, end: 0 };
+}
 export async function generateExternSignatures(
   host: CompilerHost,
   libraryPath: string,
 ): Promise<readonly Diagnostic[]> {
   const diagnostics = createDiagnosticCollector();
   const pkgJson = await readPackageJson(host, libraryPath);
-  if (!pkgJson.tspMain) {
-    throw new Error("Must have a tspMain with decorator declaration.");
+  if (!pkgJson.exports) {
```

Comment:
Do we want to drop support for just using `tspMain`? I'm fine with that given this is only used internally but probably need to double check that our scaffolding is updated to use `exports`.

Alternatively - check for `tspMain` and set it as the `"."` key in the `exportsMap`?

Reply:
yeah wanted to drop it as I think we should recommend people use `exports` asnd having tspd do that would be good